From 00221a68bdf672d4539fa0c7401fa2d284fee421 Mon Sep 17 00:00:00 2001
From: Gabe Black <gabe.black@gmail.com>
Date: Mon, 31 Jan 2022 16:43:16 -0800
Subject: [PATCH 231/757] scons: Exclude parsetab.py when collecting python
 deps.

Because we don't have a good way to actually figure out what python
files we depend on, we have to use Glob and wildcard matching to depend
on all potential python files. Unfortunately that will pick up the
parsetab.py file that ply generates, which is a cached intermediate file
and not an input.

Change-Id: Id3dc0083e97c145deca04939182157623d6b780f
Reviewed-on: https://gem5-review.googlesource.com/c/public/gem5/+/56341
Maintainer: Gabe Black <gabe.black@gmail.com>
Tested-by: kokoro <noreply+kokoro@google.com>
Reviewed-by: Jason Lowe-Power <power.jg@gmail.com>
---
 src/arch/SConscript | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/arch/SConscript b/src/arch/SConscript
index 3034dac..0178750 100644
--- a/src/arch/SConscript
+++ b/src/arch/SConscript
@@ -104,7 +104,8 @@ SCons.Tool.SourceFileScanner.add_scanner('.cc.inc', SCons.Tool.CScanner)
 # output from the ISA description (*.isa) files.
 #
 
-parser_files = Glob('isa_parser/*.py')
+# parsetab.py is a file generated by PLY which we don't want to add as a dep.
+parser_files = Glob('isa_parser/*.py', exclude=['*/parsetab.py'])
 micro_asm_py = File('micro_asm.py')
 
 # import ply here because SCons screws with sys.path when performing actions.
-- 
1.8.3.1

